trust/meson.build

```diff
@@ -0,0 +1,165 @@
+libtrust_data_sources = [
+  'asn1.c', 'asn1.h',
+  'base64.c', 'base64.h',
+  'pem.c', 'pem.h',
+  'oid.c', 'oid.h',
+  'utf8.c', 'utf8.h',
+  'x509.c', 'x509.h'
+]
+
+asn_sources = [
+  'basic.asn',
+  'pkix.asn',
+  'openssl.asn'
+]
+
+asn_h_deps = []
+foreach name : asn_sources
+  asn_h = custom_target('generate @0@.h'.format(name),
+                        output: '@0@.h'.format(name),
+                        input: name,
+                        command: [
+                          asn1Parser, '-o', '@OUTPUT@', '@INPUT@'
+                        ])
+  asn_h_dep = declare_dependency(sources: asn_h)
```

Comment:
I wouldn't use an internal dependency just for source files. My reasoning behind internal dependencies, in a similar way to external dependencies, the usually hold linking parameters (libraries), header directories (include directories), compiler flags and sometimes also sources, so I use them just to hold generated libraries among with their include directories and external dependencies and also generated source files. When grouping source files I just use the `files` function or just an array.

Reply:
Indeed, switched to listing it in sources.